README.md

```diff
@@ -507,11 +507,19 @@ Configurations:
 
 Command | Testing TypeScript
 ---     | ---
-default | Run through PhantomJS one time with no file watching
-headless (aliases: hl, h)| Run through PhantomJS with files being watched & tests automatically re-run
-watch (alias: w)| Run through Chrome with files being watched & tests automatically re-run
-
-Note that Chrome still requires a manual refresh on the Debug tab to see updated test results.
+default | Run through defined browsers one time with no file watching
```

Comment:
I'm starting to see a conflict that I'm not sure how to reconcile. So I'll present that and we can decide how to proceed.

When I began trying to understand how client-side testing should should work, PhantomJS was there to help fill that gap needed for headless testing. But because it's not an actual browser, some implementations didn't work as expected in real browsers. Since then, browser makers have added facilities to allow developers to use actual browsers during tests which also gave rise to Selenium Grid providers like BrowserStack, SauceLabs and others. Most, if not all offer a free version of their service of some kind. This further pushes the need to use PhantomJS in the background.

This is where I was when I suggested these changes. I actually use SauceLabs for my project and don't want to use PhantomJS at all on my CI server in case something PhantomJS doesn't support well fails. (In my case PhantomJS is fine but I was thinking of the broader community). At this time, there isn't a way to only use the browsers I configure in the Karma config. We're forced to use PhantomJS if we don't have any parameters or if we specify headless.

Now, when angular-librarian creates a new project for a developer, it sets the browsers property to Chrome and PhantomJS. Default as I've written will run these two browsers if the developers doesn't do anything which fulfills the convention over configuration issue you presented but exposes a different problem with the publish. 

I also use TravisCI since it was free with GitHub useage. CI systems usually have a few phases like, checkout, build, test, and "post testing". Typically, after testing succeeds, you may want to publish your library and that's where the "post testing" phase comes in. The current setup will cause another test to occur just because I'm publishing. It seems like some of the decisions made for the angular-librarian workflow were based on a developers doing everything from their computer. I'm not actually saying that was how it was done but I had to heavily modify some of the files angular-librarian generated to get it to allow the CI server to work as I expected it to.

angular-librarian should ideally make it easy to handle both situations. Since I'm familiar with the CI half, I've been trying to add that functionality to it. Maybe we should look at these changes more holistically to see how it fits in overall.

Reply:
Yeah, I think, perhaps, we can make it so that if no browsers are provided, either via the command line or by a config file, we fallback to the defaults of PhantomJS or Chrome. Does that sound ok to you?

Reply:
Ok just so I'm not making assumptions about what you're saying I'm going to paraphrase what I believe you are saying. angular-librarian will still by default place Chrome and PhantomJS in the browsers property. The default testing option of not specifying any parameters will provide the new functionality of this PR, which is to run the browsers specified in the karma file. But if the developer has purposefully removed all browsers from the karma file, the tool will attempt to at the minimum use PhantomJS or Chrome. Is my assessment correct or did I miss anything? If this is correct, I'd like to suggest just using PhantomJS just in case a CI server is involved, since CI servers generally don't have browsers installed.

Reply:
Glad you asked to clarify! Here's the scenarios I'm thinking of...(note that `ngl test` is the same as running `npm test`)

1) User runs the following:
    ```shell
    npm test
    ```
    Since no browser overrides have been provided, it falls back to PhantomJS

2) User provides a command-line argument for `ChromeHeadless`:
    ```shell
    npm test --browsers=ChromeHeadless
    ```
    Since the custom config provides an override, tests are run with `ChromeHeadless`
4) User provides a command-line argument for Chrome & Firefox for a watch
    ```shell
    npm test watch --browsers=Chrome,Firefox
    ```
    Both Chrome & Firefox are opened by Karma and files are watched

However, does this cover your use case for CI testing? I'm wondering if there needs to be a way of specifying browsers to fallback to as either a custom config or on project init.

Reply:
Also, I really want to thank you for the effort you've already put in between discussions and actual code!

Reply:
Yeah, no problem but I have to admit helping is slightly selfish on my part. I'm hoping that the library can get to the point where I won't have to do a lot of work every time there's an update. :) lol.

Anyway, I think you're over thinking this a bit. The karma file should be the source of truth for configuration. If a developer is running Karma tests, there's supposed to be browsers there for the tests to use. The override is for when you want to use a sub or super set of those browsers. In my specific case, my karma file has 10 browsers for SauceLabs testing. When I run the testing, I want those browsers to be used and not added to or overridden with PhantomJS.

On the other hand, if the karma file itself specifies no browsers by the time the library is ready to start Karma THEN we fallback to PhantomJS because a Karma test with no browser is completely useless.

Reply:
Ok, I'm ok with the proposal if there's no browsers specified always falling back to PhantomJS.